### PR TITLE
NR-249372: Automate prereleases to releases

### PR DESCRIPTION
**Pull request by marcsanmi**
This PR updates and renames the `trigger_prerelease.yaml` workflow to `automated_release.yaml`, to both promote appropriate prereleases to full releases and trigger the creation of new prereleases.

~~**Note:**  Do not merge until this other PR https://github.com/newrelic/coreint-automation/pull/63 is merged.~~

### CODE REVIEW

**Comment by paologallinaharbur** [comment]
I've moved it to draft since it should not be merged

**Review by paologallinaharbur — APPROVED** [review]

File: .github/workflows/automated_release.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Automated release creation
+
+# This workflow triggers a centralized reusable workflow for managing the entire release process, which:
+# 1. Checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure",
+#    and promotes it to a full release if found suitable.
+# 2. Automatically triggers the creation of a new prerelease based on the specified criteria, fully automating the release cycle.
+#
+# The workflow is designed to be triggered from the default branch, supporting both automated and manual prerelease
+# creations for flexibility in release management.
```
Comment (paologallinaharbur) [inline .github/workflows/automated_release.yaml:9]: is this text needed?

Comment (sigilioso) [reply .github/workflows/automated_release.yaml:9]: We could also _centralise_ this documentation by adding a link (only if we really think it is useful, considering the action being used below, it is pretty obvious): https://github.com/newrelic/coreint-automation/blob/main/.github/workflows/reusable_release_automation.yaml#L3-L9

Comment (marcsanmi) [reply .github/workflows/automated_release.yaml:9]: I added it for extra context and not having to jump into the reusable... but it's indeed explaining the reusable itself. I'm going to remove it.

**Pull request merged**